corehq/apps/userreports/reports/filters/values.py

```diff
@@ -128,35 +133,129 @@ def to_sql_values(self):
         }
 
 
+def get_dyn_range_filter(date_range_slug):
+
+    class DynRangeFilter(BasicFilter):
+        date_range = date_range_slug
+
+        def build_expression(self, table):
+            from sqlalchemy import bindparam
+            start_date, end_date = get_daterange_start_end_dates(self.date_range, *self.parameter)
+            return get_column(table, self.column_name).between(
+                bindparam(start_date), bindparam(end_date)
+            )
+
+    return DynRangeFilter
```

Comment:
does the class need to be returned dynamically as opposed to instantiated for some reason? again, not necessarily saying needs to be changed but it's an atypical pattern